pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -148,16 +151,25 @@ func (p *Provider) createAMIOptions(ctx context.Context, nodeTemplate *v1alpha1.
 	if len(securityGroupsIDs) == 0 {
 		return nil, fmt.Errorf("no security groups exist given constraints")
 	}
+	if err != nil {
+		return nil, err
+	}
+	anyPublicIPv4Associations, err := p.subnetProvider.CheckAnyPublicIPv4Associations(ctx, nodeTemplate)
+	if err != nil {
+		return nil, err
+	}
+
 	return &amifamily.Options{
-		ClusterName:             settings.FromContext(ctx).ClusterName,
-		ClusterEndpoint:         p.ClusterEndpoint,
-		AWSENILimitedPodDensity: settings.FromContext(ctx).EnableENILimitedPodDensity,
-		InstanceProfile:         instanceProfile,
-		SecurityGroupsIDs:       securityGroupsIDs,
-		Tags:                    tags,
-		Labels:                  labels,
-		CABundle:                p.caBundle,
-		KubeDNSIP:               p.KubeDNSIP,
+		ClusterName:                settings.FromContext(ctx).ClusterName,
+		ClusterEndpoint:            p.ClusterEndpoint,
+		AWSENILimitedPodDensity:    settings.FromContext(ctx).EnableENILimitedPodDensity,
+		InstanceProfile:            instanceProfile,
+		SecurityGroupsIDs:          securityGroupsIDs,
+		AssociatePublicIpv4Address: aws.Bool(anyPublicIPv4Associations),
```

Comment:
```suggestion
		AssociatePublicIpAddress: aws.Bool(anyPublicIPv4Associations),
```

We should probably align this name with the API